src/commands/case_management.py

```
@@ -0,0 +1,700 @@
+
+    # We either have a valid case or we've left the method at this point.
+    async with ctx.bot.board.modify_rescue(rescue) as case:
+        case.active = not case.active
+        await ctx.reply(f'{case.client}\'s case is now {"Active" if case.active else "Inactive"}.')
+
+
+@require_channel
+@require_permission(RAT)
+@command("assign", "add", "go")
+async def cmd_case_management_assign(ctx: Context):
+    if len(ctx.words) <= 2:
+        await ctx.reply("Usage: !assign <Client Name|Case Number> <Rat 1> <Rat 2> <Rat 3>")
+        return
+
+    # Pass case to validator, return a case if found or None
+    rescue = _validate(ctx, ctx.words[1])
+
+    if not rescue:
+        await ctx.reply("No case with that name or number.")
+        return
+
+    # Get rats from input command
+    rat_list = ctx.words_eol[2].split()
+
+    # Get client's IRC nick, otherwise use client name as entered
+    rescue_client = rescue.irc_nickname if rescue.irc_nickname else rescue.client
+
+    async with ctx.bot.board.modify_rescue(rescue.board_index) as case:
+        logger.debug("assigning {!r} to case {}", rat_list, rescue.board_index)
+        for name in rat_list:
+            rat = Rat(name=name, uuid=None)
+            await case.add_rat(rat)
+
+            if rat.name in case.unidentified_rats:
+                await ctx.reply(f"Warning: {name!r} is NOT identified.")
+
+    await ctx.reply(f'{rescue_client}: Please add the following rat(s) to your friends list:'
+                    f' {", ".join(str(rat) for rat in rat_list)}')
+
+
+@require_channel
+@require_permission(RAT)
+@command("clear", "close")
+async def cmd_case_management_clear(ctx: Context):
+    if len(ctx.words) < 2:
+        await ctx.reply("Usage: !clear <Client Name|Board Index> [First Limpet Sender]")
+        return
+
+    # Pass case to validator, return a case if found or None
+    rescue = _validate(ctx, ctx.words[1])
+
+    # Only set First Limpet if it was specified.
+    if len(ctx.words) == 3:
```

[Question] Maybe we should throw a usage error if the command has more than 3 words, too? Just to be safe. e.g. `!close 3 unknown [pc]` would close the case with no FL.

---

yeah, this one would be better to throw a usage error; you can't close a case to more than one FL. After all, its not called a _first_ limpet for nothing.